pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2677,29 +2687,55 @@ func validateAppNetworkConfig(ctx *zedrouterContext, appNetConfig types.AppNetwo
 	if len(ulCfgList0) == 0 {
 		return true
 	}
-	sub := ctx.subAppNetworkConfig
-	items := sub.GetAll()
-	for _, cfg := range items {
-		appNetConfig1 := cast.CastAppNetworkConfig(cfg)
-		ulCfgList1 := appNetConfig1.UnderlayNetworkList
+	if containsACLPortMapRule(ctx, ulCfgList0) {
+		log.Errorf("app (%s) on network with no uplink and has portmap rule\n",
+			appNetConfig.DisplayName)
+		errStr := fmt.Sprintf("network with no uplink, has portmap")
+		err := errors.New(errStr)
+		addError(ctx, appNetStatus, "underlayACL", err)
+		return false
+	}
+	pub := ctx.pubAppNetworkStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		appNetStatus1 := cast.CastAppNetworkStatus(st)
+		ulCfgList1 := appNetStatus1.UnderlayNetworkList
 		// XXX can an delete+add of app instance with same
 		// portmap result in a failure?
-		if appNetConfig.DisplayName == appNetConfig1.DisplayName ||
-			len(ulCfgList1) == 0 {
+		if appNetStatus.DisplayName == appNetStatus1.DisplayName ||
+			appNetStatus1.Error != "" || len(ulCfgList1) == 0 {
 			continue
 		}
 		if checkUnderlayNetworkForPortMapOverlap(ctx, appNetStatus, ulCfgList0, ulCfgList1) {
 			log.Errorf("app %s and %s have duplicate portmaps",
-				appNetConfig.DisplayName, appNetConfig1.DisplayName)
+				appNetStatus.DisplayName, appNetStatus1.DisplayName)
 			return false
 		}
 	}
 	return true
 }
 
+// whether there is a portmap rule, on with a network instance with no
+// uplink interface
+func containsACLPortMapRule(ctx *zedrouterContext,
```

Comment:
Name confused me.
contailsDetachedACLPortMapRule() or something similar would be less confusing.

Reply:
sure. done.